tests/middleware/test_http.py

```diff
@@ -247,43 +259,24 @@ async def dispatch(
         client.get("/")
 
 
-ctxvar: contextvars.ContextVar[str] = contextvars.ContextVar("ctxvar")
-
-
-class PureASGICustomMiddleware:
-    def __init__(self, app: ASGIApp) -> None:
-        self.app = app
+def test_modify_content_type(
+    test_client_factory: Callable[[ASGIApp], TestClient]
+) -> None:
+    async def dispatch(request: HTTPConnection) -> AsyncGenerator[None, Response]:
+        resp = yield
+        resp.media_type = "text/csv"
```

Comment:
I'm not sure we need to be that defensive, to be honest. If we're clear enough in the documentation (which ought to be written before we consider merging this anyway, right?) that this is a "simplified response interface, allowing to inspect the status code and headers, or tweak headers", then maybe that's enough.

Though, now that I think of it — what if users _do_ want to inspect `response.body` in the "on response" part of the dispatch flow? Right now, it's empty, and actually makes no sense because this is called before we start processing any chunk of the body. Hmm.

Reply:
Yeah I don't think there's any behavior that's right or wrong here, it's just about what will be least confusing.